plutus-core/untyped-plutus-core/test/Transform/Simplify.hs

```diff
@@ -227,6 +227,18 @@ multiApp = runQuote $ do
       app = mkIterAppNoAnn lam [mkConstant @Integer () 1, mkConstant @Integer () 2, mkConstant @Integer () 3]
   pure app
 
+forceDelayNoApps :: Term Name PLC.DefaultUni PLC.DefaultFun ()
```

Comment:
btw you can write these with input files that have program syntax in them. That can be easier for simple programs like this! I'm not sure why this file has so many test cases defined by explicitly constructing ASTs...

Reply:
Oh thanks for the tip! I can't seem to find any example, though, could you share one?

Reply:
Hmm, weird, there just aren't any for UPLC? Not sure why. But look at e.g. `PlutusIR.Transform.Beta.Test`

Reply:
Yeah UPLC tests have always been done this way. The lack of types makes it easy to construct terms in Haskell. I find it at least as easy as writing UPLC by hand.